src/Compilers/Core/Portable/StrongName/CryptoBlobParser.cs

```diff
@@ -225,6 +233,17 @@ public unsafe static bool TryGetPublicKey(ImmutableArray<byte> blob, out Immutab
                 // The RSA magic key must match the blob id
                 if (type == PrivateKeyBlobId && magic == RSA2)
                 {
+                    RSAParameters privKey = default;
+                    privKey.Exponent = expBytes;
+                    privKey.Modulus = pubKeyData;
+                    privKey.P = blobReader.ReadBytes(modulusLength / 2);
```

Comment:
If the SnPublicKey format is the CAPI public blob then

```C#
using (RSACryptoServiceProvider rsa = new RSACryptoServiceProvider())
{
    rsa.ImportParameters(rsaParameters);
    byte[] publicCapiBlob = rsa.ExportCspBlob(false);
}
```

If it's some other format: no.